### PR TITLE
Enhance Settings UI and Add Versioning

### PR DESCRIPTION
**Pull request by ItsSimko**
## Summary by Sourcery

Enhance the user interface of the Markdown and Settings windows by improving layout management and error handling. Introduce versioning by creating a version text file during the CI release process. Adjust the Settings window to dynamically center itself and improve the layout of UI elements for better user experience.

Enhancements:
- Improve the MarkdownWindow class by adding error handling for file not found scenarios and restructuring the UI layout for better usability.
- Enhance the SettingsWindowUI by adjusting the window size and centering it relative to the parent window, and reorganizing the UI elements for better layout management.
- Add version to settings.txt and bottom of settings window. Version is automatically inserted on github actions release tag and uses the tags version.

CI:
- Add a step in the release workflow to create a version text file for PyInstaller, ensuring version consistency across builds.

### CODE REVIEW

**Comment by ItsSimko** [comment]
@sourcery-ai review

File: src/FreeScribe.client/UI/SettingsWindowUI.py [hunk 1]
```diff
@@ -194,23 +194,23 @@ def create_whisper_settings(self):
...
+        self.whisper_architecture_label.grid(row=left_row, column=0, padx=0, pady=5, sticky="w")
         whisper_architecture_options = self.settings.get_available_architectures()
-        self.whisper_architecture_dropdown = ttk.Combobox(right_frame, values=whisper_architecture_options, width=20, state="readonly")
+        self.whisper_architecture_dropdown = ttk.Combobox(left_frame, values=whisper_architecture_options, width=20, state="readonly")
         if self.settings.editable_settings[SettingsKeys.WHISPER_ARCHITECTURE.value] in whisper_architecture_options:
             self.whisper_architecture_dropdown.current(whisper_architecture_options.index(self.settings.editable_settings[SettingsKeys.WHISPER_ARCHITECTURE.value]))
         else:
             # Default cpu
-            self.whisper_architecture_dropdown.set()
+            self.whisper_architecture_dropdown.set("CPU")
```
[the remark this answers is not in the document]
Comment (ItsSimko) [reply src/FreeScribe.client/UI/SettingsWindowUI.py]: https://github.com/ClinicianFOCUS/FreeScribe/pull/240/commits/32028fa73012c771570915c19fc9ee50a4eb455f

**Comment by ItsSimko** [comment]
@sourcery-ai review

File: src/FreeScribe.client/UI/SettingsWindowUI.py [hunk 2]
```diff
@@ -194,23 +211,23 @@ def create_whisper_settings(self):
         self.settings.editable_settings_entries["Whisper Model"] = self.whisper_models_drop_down
 
         # create the whisper model dropdown slection
-        microphone_select = MicrophoneSelector(left_frame, left_row, 0, self.settings)
+        microphone_select = MicrophoneSelector(right_frame, right_row, 0, self.settings)
```
[the remark this answers is not in the document]
Comment (ItsSimko) [reply src/FreeScribe.client/UI/SettingsWindowUI.py:214]: https://github.com/ClinicianFOCUS/FreeScribe/pull/240/commits/c05f159ad4fa7e46541eb076a4db4fc003b12964

File: src/FreeScribe.client/UI/SettingsWindow.py [hunk 3]
```diff
@@ -607,4 +610,14 @@ def update_whisper_model(self):
                 old_whisper_architecture != self.editable_settings_entries[SettingsKeys.WHISPER_ARCHITECTURE.value].get() or 
                 old_cpu_count != self.editable_settings_entries[SettingsKeys.WHISPER_CPU_COUNT.value].get() or
                 old_compute_type != self.editable_settings_entries[SettingsKeys.WHISPER_COMPUTE_TYPE.value].get()):
-            self.main_window.root.event_generate("<<LoadSttModel>>")
\ No newline at end of file
+            self.main_window.root.event_generate("<<LoadSttModel>>")
+
+    def get_application_version(self):
+        version_str = "vx.x.x.alpha"
+        try:
```
[the remark this answers is not in the document]
Comment (ItsSimko) [reply src/FreeScribe.client/UI/SettingsWindow.py:617]: https://github.com/ClinicianFOCUS/FreeScribe/pull/240/commits/e8ae8cd76b40245a569a64a42f3ee732ff8a6107

**Review by yingbull — APPROVED** [review]

**Pull request merged**